Automattic/wp-calypso
# Add Instagram to post list preview

Completes 1203820933396971-as-1204692224839572/f

## Proposed Changes

* This PR adds Instagram preview to the post list share preview for the blogs that have Instagram connection enabled

## Testing Instructions

- Ensure that you have D111103-code applied to your sandbox
- Add the blog sticker to your site as mentioned in D111103-code
- Point `public-api.wordpress.com` to your sandbox
- Goto `/posts/:site` 
- Click on the 3 dots for any post _that has an image_ and Click "Share"
- Click on "Preview"
- Confirm that you see the Instagram preview
- Now do the same for a post that doesn't have an image
- Confirm that you see a notice that says, "You need a valid image in your post to share to Instagram."
- Confirm that you do not see the Instagram preview for a site that doesn't have Instagram enabled.

Post with an image

<img width="1304" alt="Screenshot 2023-06-06 at 12 37 19 PM" src="https://github.com/Automattic/wp-calypso/assets/18226415/056639cb-2085-490d-a65a-b45f4e5e6cf9">

Post without an image

<img width="1308" alt="Screenshot 2023-06-06 at 12 37 44 PM" src="https://github.com/Automattic/wp-calypso/assets/18226415/6bc5aebc-6967-4ea2-867f-dd237dd14f4c">


## Pre-merge Checklist
